controllers/knative/knative.go

```diff
@@ -84,3 +102,100 @@ func GetKnativeAvailability(cfg *rest.Config) (*Availability, error) {
 		return result, nil
 	}
 }
+
+func GetWorkflowSink(ctx context.Context, c client.Client, workflow *operatorapi.SonataFlow, pl *operatorapi.SonataFlowPlatform) (*duckv1.Destination, error) {
+	if workflow == nil {
+		return nil, nil
+	}
+	if workflow.Spec.Sink != nil {
+		return workflow.Spec.Sink, nil
+	}
+	if pl != nil {
+		if pl.Spec.Eventing != nil {
+			// no sink defined in the workflow, use the platform broker
+			return pl.Spec.Eventing.Broker, nil
+		} else if pl.Status.ClusterPlatformRef != nil {
+			// Find the platform referred by the cluster platform
+			platform := &operatorapi.SonataFlowPlatform{}
+			if err := c.Get(ctx, types.NamespacedName{Namespace: pl.Status.ClusterPlatformRef.PlatformRef.Namespace, Name: pl.Status.ClusterPlatformRef.PlatformRef.Name}, platform); err != nil {
```

Comment:
Don't we already have a function to read the platform doing a similar logic? We pass the `SonataFlowPlatform` and get back either itself or the cluster ref?